src/components/generic/DropdownSelect/DropdownSelect.jsx

```diff
@@ -0,0 +1,91 @@
+import React, {useState, useEffect} from 'react';
+import PropTypes from 'prop-types';
+
+import Icon from '../Icon/Icon';
+import OptionsList from '../OptionsList/OptionsList';
+import webexComponentClasses from '../../helpers';
+
+/**
+ * DropdownSelect Component
+ *
+ * @param {object} props  Data passed to the component
+ * @param {string} props.className  Custom CSS class to apply
+ * @param {string} props.value  Selected option
+ * @param {object[]} props.options  Array of options
+ * @param {Function} props.onClick  Extra action to perform on select item click
+ * @param {boolean} props.disabled  The select state
+ * @returns {object}  JSX of the element
+ */
+export default function Dropdownselect({
+  className,
+  value,
+  options,
+  onClick,
+  disabled,
+}) {
+  const [selectedText, setSelectedText] = useState(value);
+  const [showOptionList, setShowOptionList] = useState(false);
+  const cssClasses = webexComponentClasses(`dropdown-select ${className} ${disabled ? 'disabled' : ''}`);
+
+  const handleListDisplay = () => {
+    if (!disabled) {
+      setShowOptionList(!showOptionList);
+    }
+  };
+
+  const handleOptionClick = (option) => {
+    setSelectedText(option.label);
+    setShowOptionList(false);
+    onClick(option.value);
+  };
+
+  useEffect(() => {
+    const valueIndex = options && options.findIndex((option) => option.value === value);
+
+    const newValue = valueIndex !== null && valueIndex !== -1 ? options[valueIndex].label : value;
+
+    setSelectedText(newValue);
+  }, [value, options]);
+
+  return (
+    <div className={cssClasses}>
+      <div
+        className={`selected-option ${showOptionList ? 'active' : ''}`}
```

Comment:
let's have a single terminology for the 'options shown' state, we currently have 'show' and 'active' meaning the same thing. I would suggest to use the word 'expanded'.